rest_framework/fields.py

```diff
@@ -1270,6 +1274,70 @@ def to_internal_value(self, data):
         return data
 
 
+class RecursiveField(Field):
+    """
+    A field that gets its representation from its parent.
+
+    This method could be used to serialize a tree structure, a linked list, or
+    even a directed acyclic graph. As with all recursive things, it is
+    important to keep the base case in mind. In the case of the tree serializer
+    example below, the base case is a node with an empty list of children. In
+    the case of the list serializer below, the base case is when `next==None`.
+    Above all, beware of cyclical references.
+
+    Examples:
+
+    class TreeSerializer(self):
+        children = ListField(child=RecursiveField())
+
+    class ListSerializer(self):
+        next = RecursiveField(allow_null=True)
+    """
+
+    def __init__(self, to='self', to_module=None, **kwargs):
+        self.to = to
+        self.to_module = to_module
+        field_kwargs = dict(
+            (key, kwargs[key])
+            for key in kwargs
+            if key in inspect.getargspec(Field.__init__)
+        )
+        super(RecursiveField, self).__init__(**field_kwargs)
+
+    def bind(self, field_name, parent):
+        super(RecursiveField, self).bind(field_name, parent)
+
+        if hasattr(parent, 'child') and parent.child is self:
+            parent_class = parent.parent.__class__
+        else:
+            parent_class = parent.__class__
+
+        if self.to == 'self':
+            proxy_class = parent_class
+        else:
+            ref = importlib.import_module(self.to_module or parent_class.__module__)
```

Comment:
We could reuse `import_from_string` from here... https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/settings.py#L150